p2p/test/transport/transport_test.go

```diff
@@ -688,3 +692,34 @@ func TestDiscoverPeerIDFromSecurityNegotiation(t *testing.T) {
 		})
 	}
 }
+
+// TestCloseConnWhenBlocked tests that the server closes the connection when the rcmgr blocks it.
+func TestCloseConnWhenBlocked(t *testing.T) {
+	for _, tc := range transportsToTest {
+		if tc.Name == "WebRTC" {
+			continue // WebRTC doesn't have a connection when we block so there's nothing to close
```

Comment:
Should we test this, `NewStream` on client will still fail, which is what we want. It'll fail with `DeadlineExceeded` though

Reply:
I don't think we should. It's not testing what we think. The only thing it tests is whether the server ignores us and we hit a deadline.

Reply:
Maybe we could make another tests that just focuses on that behavior for WebRTC though?

Reply:
It would test that if the respurce manager blocks no connection is established. But we can handle that separately. I am fine with keeping it as is.

Reply:
#2856